### PR TITLE
[PKG-3053] Upgrade to 8.4.0

### PR DESCRIPTION
**Pull request by rafaelmartins**
fixes CVE-2023-38545. CVE-2023-38546.

nss and mingw (the old one, not -w64) support removed, but we don't use them.

osx patch not needed anymore, was part of 8.3.0.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,9 +19,6 @@ requirements:
     # perl is required to run the tests on UNIX.
     - perl           # [unix]
     - pkg-config     # [unix]
-    - patch          # [osx]
-    - autoconf       # [osx]
-    - automake       # [osx]
   host:
     - krb5        1.20.1
     - libnghttp2  1.52.0  # [unix]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:24]: FYI, `libnghttp2 <1.57` has the [CVE-2023-44487](https://nvd.nist.gov/vuln/detail/CVE-2023-44487). It's found recently. The fix is here https://github.com/nghttp2/nghttp2/pull/1961. 
If it's possible we should update it ASAP.

Comment (rafaelmartins) [reply recipe/meta.yaml:24]: AFAIU, this cve is related to server implementation of http2, which curl does not relies on, but I'll bump it too anyway.

Comment (skupr-anaconda) [reply recipe/meta.yaml:24]: Ah, thanks for the explanation!

Comment (skupr-anaconda) [reply recipe/meta.yaml:24]: `libnghttp2 1.57.0` is already on defaults

**Review by skupr-anaconda — APPROVED** [review]
LGTM

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

**Pull request merged**